res/controllers/Roland_DJ-505-scripts.js

```diff
@@ -0,0 +1,1405 @@
+    // 0x00 is ON, 0x01 is OFF
+    engine.setValue("[Mixer Profile]", "xFaderReverse", (value == 0x00) ? 1 : 0);
+};
+
+DJ505.setChannelInput = function (channel, control, value, status, group) {
+    var number = (channel == 0x00) ? 0 : 1;
+    var channelgroup = "[Channel" + (number + 1) + "]";
+    switch(value) {
+    case 0x00:  // PC
+        engine.setValue(channelgroup, "passthrough", 0);
+        break;
+    case 0x01:  // LINE
+    case 0x02:  // PHONO
+        engine.setValue(channelgroup, "passthrough", 1);
+        break;
+    }
+};
+
+DJ505.Deck = function (deckNumbers, offset) {
+    components.Deck.call(this, deckNumbers);
+
+    this.slipModeButton = new DJ505.SlipModeButton({
+        midi: [0x90 + offset, 0xF],
+        shiftOffset: -8,
+        shiftControl: true,
+        sendShifted: true,
+    });
+
+    engine.setValue(this.currentDeck, "rate_dir", -1);
+    this.tempoFader = new components.Pot({
+        group: "[Channel" + deckNumbers + "]",
+        midi: [0xB0 + offset, 0x09],
+        connect: function () {
+            engine.softTakeover(this.group, "pitch", true);
+            engine.softTakeover(this.group, "rate", true);
+            components.Pot.prototype.connect.apply(this, arguments);
+        },
+        unshift: function () {
+            this.inKey = "rate";
+            this.inSetParameter = components.Pot.prototype.inSetParameter;
+            engine.softTakeoverIgnoreNextValue(this.group, "pitch");
+        },
+        shift: function () {
+            this.inKey = "pitch";
+            this.inSetParameter = function (value) {
+                // Scale to interval ]-7…7[; invert direction as per controller
+                // labeling.
+                value = 14 * value - 7;
+                value *= -1;
+                components.Pot.prototype.inSetValue.call(this, value);
+            };
+            engine.softTakeoverIgnoreNextValue(this.group, "rate");
+        }
+    });
+
+
+    // ============================= JOG WHEELS =================================
+    this.wheelTouch = function (channel, control, value, status, group) {
+        if (value === 0x7F && !this.isShifted) {
+            var alpha = 1.0/8;
+            var beta = alpha/32;
+            engine.scratchEnable(script.deckFromGroup(this.currentDeck), 512, 45, alpha, beta);
+        } else {    // If button up
+            engine.scratchDisable(script.deckFromGroup(this.currentDeck));
+        }
+    };
+
+    this.wheelTurn = function (channel, control, value, status, group) {
+        var newValue = value - 64;
+        var deck = script.deckFromGroup(this.currentDeck);
+        if (engine.isScratching(deck)) {
+            engine.scratchTick(deck, newValue); // Scratch!
+        } else if (this.isShifted) {
+            // Strip search.
+            var oldPos = engine.getValue(this.currentDeck, "playposition");
+            // Scale to interval [0,1].
+            newValue = newValue / 0xff;
```

Comment:
The delta newValue will be in the range [-64, 63]. Dividing by 255 will not scale it to the unit interval as the comment pretends and is confusing.

Using just DJ505.stripSearchScaling as the scaling constant is sufficient. 50/255 equals ~0.2. Scaling by the actual duration is still needed to provide a linear feedback between motion/rotation and time.

Reply:
Fixed in 31b62638cc7c9491548cbe0b70d38593d9aa5e5a. I used 0.15 instead of 0.2 as stripSearchScaling constant because that provides a bit more precision when searching through the track.

Reply:
Pinging @Lykos123 and @upCASE because that issue is also present in the mapping for the Roland DJ-202 (original PR #1595, [updated version gist](https://gist.github.com/upCASE/88fb258e85546092af18bc3ff924f752#file-roland_dj-202-scripts-js)).

Reply:
Thanks! Will fix this in my version and see if I find the time to finally upload this and create a PR for the 202 mapping.